nixos/modules/tasks/filesystems/zfs.nix

```
@@ -515,7 +519,20 @@ in
               done
               poolImported "${pool}" || poolImport "${pool}"  # Try one last time, e.g. to import a degraded pool.
               if poolImported "${pool}"; then
-                ${optionalString cfgZfs.requestEncryptionCredentials "\"${packages.zfsUser}/sbin/zfs\" load-key -r \"${pool}\""}
+                ${optionalString cfgZfs.requestEncryptionCredentials ''
+                  ${packages.zfsUser}/sbin/zfs list -rHo name,keylocation ${pool} | while read ds kl; do
```

This read isn't safe for all valid dataset names, but the fix is easy:

```suggestion
                  ${packages.zfsUser}/sbin/zfs list -rHo name,keylocation ${pool} | while IFS=$'\t' read ds kl; do
```

```
[grahamc@Petunia:~]$ zfs list -rHo name,keylocation tank/local/junk
tank/local/junk	none
tank/local/junk/foo bar	none
tank/local/junk/foo: ...b-a_r	none
tank/local/junk/foo: ...bar	none
```

The original behavior of this script would stop reading at the ` `:
```
[grahamc@Petunia:~]$ ./keyloc.sh
load-key tank/local/junk/foo
load-key tank/local/junk/foo:
load-key tank/local/junk/foo:
```

but with the IFS:

```
zfs list -rHo name,keylocation tank/local/junk | while IFS=$'\t' read ds kl; do
    echo "$ds"
    echo "$kl"
done
```

produces:

```
tank/local/junk
none
tank/local/junk/foo bar
none
tank/local/junk/foo: ...b-a_r
none
tank/local/junk/foo: ...bar
none
```

ref: https://github.com/openzfs/zfs/blob/master/module/zcommon/zfs_namecheck.c#L57-L64
ref: https://github.com/openzfs/zfs/blob/master/module/zcommon/zfs_namecheck.c#L166-L294